docs/index.md

```diff
@@ -619,6 +619,24 @@ class UserComponent < ViewComponent::Base
 end
 ```
 
+### Handling errors
+
+An exception occurring in a small fragment of the user interface can potentially break an entire page.
+
+To prevent this you can use `rescue_from`, just like in a controller:
+
+```ruby
+class FailingComponent < ViewComponent::Base
+  rescue_from User::NotAuthorized, with: :deny_access
```

Comment:
I wonder if this should be `with_component` instead of `with`. That way users can define, and re-use a common error component. Thoughts?

Reply:
Not a bad idea, although I'm piggybacking `ActiveSupport::Rescuable` here. Implementing that keyword would mean a second override, and possibly a dynamic method definition:

`rescue_from Something, with_component: FallbackComponent` would create a method like this:

```ruby
def __rescue_something(exc)
  render FallbackComponent.new(exception: exc, source: self)
end
```

And then translate `with_component: FallbackComponent` to `with: :__rescue_something`

Am I overkilling or not seeing a better approach here?

Honestly, I can see both keywords being useful and maybe both should be defined. How about we start integrating this PR, and I'll try to sketch something on top of this for `with_component`?

Reply:
@joelhawksley and I have given this a bit more thought, and we think the solution to this problem should be consistent with the slot API we end up finalizing in #503 

We really like the direction this is going, but we think it's worth waiting a bit before moving forward so we end up with a cohesive API.

Reply:
That makes sense. Let's wait for #503 and catch up later for a better API :)

Reply:
SlotsV2 are merged, maybe it's time to revisit this?

Most of the time, I would rather use a `with` keyword that points to a local method, than rendering another component using `with_component`.
I see the use case for a global error component though, so maybe the `with_component` keyword could be added in a separate PR as @francescobbo suggests?

Reply:
> Most of the time, I would rather use a `with` keyword that points to a local method, than rendering another component using `with_component`.

I ran into a case where I needed to rescue an error in a component (which should be rare, in my opinion). It was useful to have a method so I could add instrumentation and error logging. Given that, I think my proposal would be:

* Use the `rescue_from` API as shown above, where `with` can reference a method or a component (similar to how slots v2 works). 
* The handler method then behaves like a slot, as-in it will render the component or string that the method returns.

Thoughts?

Reply:
Yes, exactly, I would probably add error logging in the handling method as well.

:+1: to your proposal.